src/views/main.js

```diff
@@ -65,6 +108,7 @@ const mainHeader = data => `
         </svg>
       </button>
     </li>
+    ${userMenu(data)}
```

Comment:
I originally used the `<menu>` element above to hold both the nav button and the user menu.  So, as intended, this would look more like this:
```js
  <menu>
    <li>
      <button class='nav-toggle'>
        <svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 10 8' width='20'>
          <path d='M1 1h8M1 4h8M1 7h8' stroke='#000' stroke-width='1' stroke-linecap='round'/>
        </svg>
      </button>
    </li>
    <li>${userMenu(data)}</li>
  </menu>
```
As we discussed though, I may have gotten too caught up in semantics; `<menu>` is probably be more appropriate for your user menu.  That might look a bit more like this (note that I _think_ by using `<menu>` you can avoid all the `role='menuitem'` attributes?):
```js
const userMenu = data => `
<div class='user-menu-wrapper' tabindex='-1'>
  <button
  aria-expanded='false'
  aria-haspopup='true'
  class='user-menu-button'
  title='${getMessage('menu-button-title')}'
  >
    <img src='${data.fxaProfile?.avatar}' alt='${getMessage('menu-button-alt')}' />
  </button>
  <menu class='user-menu-list user-menu-popover' hidden>
    <li tabindex='1'>...</li>
  </menu>
</div>

...

  <ul>
    <li>
      <button class='nav-toggle'>
        <svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 10 8' width='20'>
          <path d='M1 1h8M1 4h8M1 7h8' stroke='#000' stroke-width='1' stroke-linecap='round'/>
        </svg>
      </button>
    </li>
    <li>${userMenu(data)}</li>
  </ul>
```

And finally, we could probably simplify even further by removing the top level list altogether, since it's not really a list and the items are not very similar... Taking the above and modifying it one last time:

```js
  <div>
    <button class='nav-toggle'>
      <svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 10 8' width='20'>
        <path d='M1 1h8M1 4h8M1 7h8' stroke='#000' stroke-width='1' stroke-linecap='round'/>
      </svg>
    </button>
    ${userMenu(data)}
  </div>
```

Hope my thoughts are clear, happy to chat in realtime if you have additional thoughts!

Reply:
I agree that the `menu` tag as a wrapper for both — the `nav` and `userMenu` — does not make too much sense. I applied the adjustments proposed by you and use `<menu>` for the user menu.

Just as an interesting note: `ul` vs `menu` only makes a difference syntactically and not semantically when interpreted by the browser ([MDN: `<menu>` Usage notes](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/menu#usage_notes)) anymore. The tag `menuitem` has been deprecated.